### ISSUE TITLE
yolo-pose training results highly inconsistent with the results from the validation scripts

### ISSUE THREAD

**Issue by xcn700418** [ultralytics/ultralytics#16208]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

The results from the yolo-pose training is shown below.
![屏幕截图 2024-09-11 140810](https://github.com/user-attachments/assets/9866398e-2cc4-499b-9068-4f94998f137e)
However, the results from the validation scripts is shown below.
![jt1](https://github.com/user-attachments/assets/36077e4e-b914-4014-95a1-1a185f157aa8)

The validation scripts is simply using the "model.val()"
Why pose map is so different?

**Comment by Y-T-G** [comment]
Try using the same batch size as training in `model.val()`

**Comment by Y-T-G** [comment]
Did you pass `save_hybrid` = True?

**Comment by Y-T-G** [comment]
Try passing `data=your_data.yaml` in `model.val()`

**Comment by xcn700418** [comment]
> Try using the same batch size as training in `model.val()`

Yes, I tried, the result from the validation scripts slightly changes, but it still has a huge difference with the validation from training.
My training script is:
![image](https://github.com/user-attachments/assets/b2a976bf-a20d-419c-ad52-6d329778063c)

**Comment by xcn700418** [comment]
> Try passing `data=your_data.yaml` in `model.val()`

I think there is something wrong when validating the model during training, the pose map of validation set is not supposed to be really low since the datasets is not very complicated.

**Comment by Y-T-G** [comment]
Did you check the validation plots from the runs folder? Is it correct?

**Comment by xcn700418** [comment]
> Did you check the validation plots from the runs folder? Is it correct?

The validation plots of validation batch generated from the training is quite bad as shown below(4 keypoints should be on the fore finger),
![image](https://github.com/user-attachments/assets/914edf7a-527e-4098-909c-365faf991224)
However the validation plots of validation batch generated from the validation scripts is quite good as shown below,
![image](https://github.com/user-attachments/assets/aab0d9ca-cc9e-4748-a404-573fc3a76871)
Also, the performance metrics for this two cases is also different.

**Comment by Y-T-G** [comment]
Are your images square?

**Comment by Y-T-G** [comment]
@ambitious-octopus @Laughing-q There's seems to be a scaling issue.

**Comment by Y-T-G** [comment]
I'm guessing it's due to passing `augment=True` during training.

**Comment by Y-T-G** [comment]
Reproducible.

```python
In [1]: from ultralytics import YOLO
model
In [2]: model = YOLO("yolov8n-pose.pt")

In [3]: results = model.val(data="coco8-pose.yaml", augment=True)
Ultralytics YOLOv8.2.87 🚀 Python-3.10.14 torch-2.3.1 CUDA:0 (Tesla T4, 14916MiB)
YOLOv8n-pose summary (fused): 187 layers, 3,289,964 parameters, 0 gradients, 9.2 GFLOPs

Dataset 'coco8-pose.yaml' images not found ⚠️, missing path '/datasets/coco8-pose/images/val'
Downloading https://ultralytics.com/assets/coco8-pose.zip to '/datasets/coco8-pose.zip'...
100%|██████████████████████████████████████████| 334k/334k [00:00<00:00, 86.3MB/s]
Unzipping /datasets/coco8-pose.zip to /datasets/coco8-pose...: 100%|██████████| 27
Dataset download success ✅ (0.8s), saved to /datasets

val: Scanning /datasets/coco8-pose/labels/val... 4 images, 0 backgrounds, 0 corrup
val: New cache created: /datasets/coco8-pose/labels/val.cache
                 Class     Images  Instances      Box(P          R      mAP50  mAP
                   all          4         14      0.861      0.886      0.944      0.703      0.109     0.0714     0.0128    0.00364
Speed: 0.2ms preprocess, 48.3ms inference, 0.0ms loss, 14.9ms postprocess per image
Results saved to /ultralytics/runs/pose/val

In [4]: results = model.val(data="coco8-pose.yaml", augment=False)
Ultralytics YOLOv8.2.87 🚀 Python-3.10.14 torch-2.3.1 CUDA:0 (Tesla T4, 14916MiB)
val: Scanning /datasets/coco8-pose/labels/val.cache... 4 images, 0 backgrounds, 0
                 Class     Images  Instances      Box(P          R      mAP50  mAP
                   all          4         14      0.926      0.891      0.907      0.668      0.848        0.5      0.535      0.352
Speed: 0.3ms preprocess, 7.1ms inference, 0.0ms loss, 0.9ms postprocess per image
Results saved to /ultralytics/runs/pose/val2
```

**Comment by Y-T-G** [comment]
`augment` parameter is not related to training augmentations. It's for enabling test-time augmentation.

https://docs.ultralytics.com/modes/predict/#inference-arguments

**Comment by Laughing-q** [comment]
@Y-T-G @xcn700418 guys currently `augment` is for TTA and only available for Detection task, and might get errors or incorrect results with other tasks which I recently fixed the issue by performing single-scale inference for other tasks even set `augment=True`, so currently the results for pose task with/without `augment=True` should get the same results. Please try it out with the latest version. Thanks!

**Comment by ambitious-octopus** [comment]
@Y-T-G thanks for digging into this issue. Yes, the `augment` parameter is creating some confusion here.

**Comment by Laughing-q** [comment]
However I do notice the plotted val images are not proper, maybe the low mAP values are related to this or probably it's just a plotting issue. I'll take a look at this.
![pic-240911-1605-38](https://github.com/user-attachments/assets/85328854-1a54-442f-9237-4ebb1561ab35)

**Comment by Laughing-q** [comment]
ok never mind, this plot was actually caused by my incorrect local coco8-pose labels, and the plot looks good when I redownload the coco8-pose datset. :)

**Comment by Laughing-q** [comment]
@xcn700418 Please try the training and validating again with the latest version of ultralytics package. Thanks! And if the issue still appears to you, please reach out and leave a minimal reproducible example here(better also include a minimal set of your dataset since it seems we're not able to reproduce the issue with coco8-pose dataset).
